### PR TITLE
Allow throwing custom errors from handlers

### PR DESCRIPTION
**Pull request by lundibundi**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

File: lib/channel.js [hunk 1]
```diff
@@ -136,9 +136,9 @@ class Channel {
     try {
       result = await proc.invoke(context, args);
     } catch (error) {
-      const timedout = error.message === 'Timeout reached';
-      const code = timedout ? 408 : 500;
-      this.error(code, { callId, error, pass: timedout });
+      const isTimeout = error.message === 'Timeout reached';
+      const code = isTimeout ? 408 : error.code || 500;
```
Comment (agil) [inline lib/channel.js:140]: const code = error.code || (isTimeout ? 408 : 500);

File: lib/channel.js [hunk 2]
```diff
@@ -136,29 +136,35 @@ class Channel {
...
+      const httpCode = error.httpCode || (isTimeout ? 408 : 500);
+      this.error(error.code, { callId, error, httpCode });
       return;
     } finally {
       proc.leave();
     }
-    if (typeof result === 'object' && result.constructor.name === 'Error') {
-      this.error(result.code, { callId, error: result, pass: true });
+    if (
+      result !== null &&
```
Comment (tshemsedinov) [inline lib/channel.js:147]: We do not need this check, because `null` will be covered by `typeof result` check

Comment (lundibundi) [reply lib/channel.js:147]: Can you clarify, `typeof null === 'object'`?

Comment (tshemsedinov) [reply lib/channel.js:147]: My mistake, I think we need `if (result && result.constructor && result.constructor.name === 'Error') {`

**Review by lundibundi — COMMENTED** [review]
LGTM

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metacom/commit/53056edaf3ac91879b535f5d4a04089b5951e864

**Pull request closed without merging**